libass/ass.c

```diff
@@ -59,6 +59,8 @@ static const char *const ssa_event_format =
 
 #define ASS_STYLES_ALLOC 20
 
+#define ASS_CACHE_MIN_TS(CACHE_TS, NEW_TS) CACHE_TS = FFMIN(CACHE_TS, NEW_TS)
```

Comment:
This looks excessive for me. At least I prefer function over macro:
```c
static inline update_prune_ts(long long *ts, const ASS_Event *events)
{
    *ts = FFMIN(*ts, event->Start + event->Duration);
}
```
Or maybe even
```c
static inline update_prune_ts(ASS_Track *track, const ASS_Event *events)
{
    track->parser_priv->gc_next_undisplay_ts =
        FFMIN(track->parser_priv->gc_next_undisplay_ts, event->Start + event->Duration);
}
```